scipy/stats/_distribution_infrastructure.py

```diff
@@ -0,0 +1,5082 @@
+        if self._overrides('_ccdf_formula'):
+            method = self._ccdf_formula
+        elif self._overrides('_logccdf_formula'):
+            method = self._ccdf_logexp
+        elif self._tol is _null and self._overrides('_cdf_formula'):
+            method = self._ccdf_complementarity
+        else:
+            method = self._ccdf_quadrature
+        return method
+
+    def _ccdf_formula(self, x, **kwargs):
+        return NotImplementedError(self._not_implemented)
+
+    def _ccdf_logexp(self, x, **kwargs):
+        return np.exp(self._logccdf_dispatch(x, **kwargs))
+
+    def _ccdf_complementarity(self, x, **kwargs):
+        return 1 - self._cdf_dispatch(x, **kwargs)
+
+    def _ccdf_quadrature(self, x, **kwargs):
+        _, b = self._support(**kwargs)
+        return self._quadrature(self._pdf_dispatch, limits=(x, b),
+                                kwargs=kwargs)
+
+    ## Inverse cumulative distribution functions
+
+    @_set_invalid_nan
+    def ilogcdf(self, x, *, method=None):
+        r"""Inverse of the logarithm of the cumulative distribution function
+
+        For more information about the inverse cumulative distribution function,
+        see `ContinuousDistribution.icdf`.
+
+        Parameters
+        ----------
+        x : Array
+            The argument of the inverse of the logarithm of the cumulative
+            distribution function (inverse log-CDF).
+        method : {None, 'formula', 'complementarity', 'inversion'}
+            The strategy used to evaluate the inverse log-CDF.
+            By default (``None``), the infrastructure chooses between the
+            following options.
+
+            - ``'formula'``: use a formula for the inverse log-CDF itself
+            - ``'complementarity'``: evaluate the inverse log-CCDF at the
+                                    logarithmic complement of `x`.
+            - ``'inversion'``: solve numerically for the argument at which the
+                               log-CDF is equal to `x`.
+
+            Not all `method` options are available for all distributions.
+            If the selected `method` is not available, a ``NotImplementedError``
+            will be raised.
+
+        Returns
+        -------
+        out : Array
+            The inverse log-CDF evaluated at the provided argument.
+
+        See Also
+        --------
+        ContinuousDistribution.icdf
+        ContinuousDistribution.logcdf
+
+        Notes
+        -----
+        Suppose a probability distribution has support :math:`[l, r]`. The
+        inverse log-CDF takes on its minimum value of :math:`l` at
+        :math:`x = \log(0) = -\infty` and its maximum value of :math:`r` at
+        :math:`x = \log(1) = 0`. Because the log-CDF has range
+        :math:`[-\infty, 0]`, the inverse log-CDF is only defined on the
+        negative reals; for :math:`x > 0`, ``ilogcdf`` returns ``nan``.
+
+        Occasionally it is needed to find the argument of the CDF for which
```

Comment:
Have to admit my eyes stumble on this "Occasionaly" on each re-read. Don't have good suggestions, so safe to ignore.

Reply:
Well, I do need a comma after it, if that would help?

Reply:
Done.